### PR TITLE
Support for ARM param tracking

### PR DESCRIPTION
**Pull request by JasonRShaver**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/client_factory.py [hunk 1]
```diff
@@ -102,6 +102,8 @@ def configure_common_settings(cli_ctx, client):
     command_name_suffix = ';completer-request' if cli_ctx.data['completer_active'] else ''
     client._client.add_header('CommandName',  # pylint: disable=protected-access
                               "{}{}".format(cli_ctx.data['command'], command_name_suffix))
+    client._client.add_header('ParameterSetName',  # pylint: disable=protected-access
+                              "{}".format(cli_ctx.data['safe_params']))
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/commands/client_factory.py]: this can just be `cli_ctx.data['safe_params']`

Comment (JasonRShaver) [reply src/azure-cli-core/azure/cli/core/commands/client_factory.py]: This fails trying to convert a list to string ("expected string or bytes-like object"), using format seemed like an easy way to 'cast' it.

Comment (JasonRShaver) [reply src/azure-cli-core/azure/cli/core/commands/client_factory.py]: Saw the ''.join method being more common, so switched to that.

Comment (yugangw-msft) [reply src/azure-cli-core/azure/cli/core/commands/client_factory.py]: Apologize for the wrong comment at the beginning as i don't know the data types, but please do `' '.join(cli_ctx.data['safe_params'])` as that would avoid the "[]" brought in by the str.
Also, please rebase from the latest dev which i hope would get rid of the seemingly irrelevant CI error

Comment (williexu) [reply src/azure-cli-core/azure/cli/core/commands/client_factory.py]: Ci passes now. This means something went wrong with using ''.join(). The headers were perhaps malformed.

Comment (yugangw-msft) [reply src/azure-cli-core/azure/cli/core/commands/client_factory.py]: It is a `None ref` error

**Pull request merged**